app/src/main/java/io/homeassistant/companion/android/webview/WebViewActivity.kt

```diff
@@ -240,10 +243,51 @@ class WebViewActivity : BaseActivity(), io.homeassistant.companion.android.webvi
-                            val browserIntent = Intent(Intent.ACTION_VIEW, it)
-                            startActivity(browserIntent)
-                            return true
+                        try {
+                            if (it.toString().startsWith(APP_PREFIX)) {
+                                Log.d(TAG, "Launching the app")
+                                val intent = packageManager.getLaunchIntentForPackage(
+                                    it.toString().substringAfter(APP_PREFIX)
+                                )
+                                if (intent != null) {
+                                    intent.flags = Intent.FLAG_ACTIVITY_NEW_TASK
+                                    startActivity(intent)
+                                } else {
+                                    Log.w(TAG, "No intent to launch app found, opening app store")
+                                    val marketIntent = Intent(Intent.ACTION_VIEW)
+                                    marketIntent.data = Uri.parse(
+                                        MARKET_PREFIX + it.toString().substringAfter(APP_PREFIX)
+                                    )
+                                    startActivity(marketIntent)
+                                }
+                                return true
+                            } else if (it.toString().startsWith(INTENT_PREFIX)) {
+                                Log.d(TAG, "Launching the intent")
+                                val intent =
+                                    Intent.parseUri(it.toString(), Intent.URI_INTENT_SCHEME)
+                                val intentPackage = intent.`package`?.let { it1 ->
+                                    packageManager.getLaunchIntentForPackage(
+                                        it1
+                                    )
+                                }
+                                if (intentPackage != null)
+                                    startActivity(intent)
+                                else {
+                                    Log.w(TAG, "No app found for intent prefix, opening app store")
+                                    val marketIntent = Intent(Intent.ACTION_VIEW)
+                                    marketIntent.data =
+                                        Uri.parse(MARKET_PREFIX + intent.`package`.toString())
+                                    startActivity(marketIntent)
+                                }
+                                return true
+                            } else if (!webView.url.toString().contains(it.toString())) {
+                                val browserIntent = Intent(Intent.ACTION_VIEW, it)
+                                startActivity(browserIntent)
+                                return true
+                            } else
+                                Log.w(TAG, "No unique cases found to override")
```

Comment:
Not sure this justifies a warning, should probably just be debug.

```suggestion
                                Log.d(TAG, "No unique cases found to override")
```